### PR TITLE
Incorporate orientation bias filtering into Mutect wdl. Closes #831

### LINKED ISSUE #831: Orientation bias WDL needs to be integrated into the M2 workflow WDL.
see ``scripts/unsupported/ob_filter/test_ob_filter.wdl`` and integrate the two tasks into the main line M2 WDL.

### PR DESCRIPTION
**Pull request by davidbenjamin**
Tested and works -- see /dsde/working/davidben/mutect/test_wdl.

### CODE REVIEW

**Comment by davidbenjamin** [comment]
@LeeTL1220 can you review?

File: scripts/mutect2_wdl/mutect2.json [hunk 1]
```diff
@@ -17,5 +17,6 @@
   "Mutect2.dbsnp_index": "$__dbsnp_index__",
   "Mutect2.cosmic": "$__cosmic__",
   "Mutect2.cosmic_index": "$__cosmic_index__",
-  "Mutect2.picard_jar": "$__picard_jar__"
+  "Mutect2.picard_jar": "$__picard_jar__",
+  "Mutect2.run_orientation_bias_filter": "$__run_orientation_bias_filter__"
```
Comment (LeeTL1220) [inline scripts/mutect2_wdl/mutect2.json]: If it is ultimately a boolean, please name it ``$__is_run_orientation_bias_filter__`` ...

File: scripts/mutect2_wdl/mutect2.wdl [hunk 2]
```diff
@@ -98,6 +98,47 @@ task SplitIntervals {
...
 
+task CollectSequencingArtifactMetrics {
+    String gatk4_jar
+    File bam_file
+    String output_prepend
+    File ref_fasta
+    Boolean run_orientation_bias_filter
+
+    command {
+        if [ ${run_orientation_bias_filter} = true ]; \
```
Comment (LeeTL1220) [inline scripts/mutect2_wdl/mutect2.wdl]: Why run this task at all if you are not going to do anything?  Move the conditional to the workflow logic.  Use the new WDL syntax for that.  *From now on, do not make tasks that implement the conditional logic nor should you make tasks that are no-ops*  These can be very expensive.

https://software.broadinstitute.org/wdl/userguide/topic?name=wdl-plumbing

File: scripts/mutect2_wdl/mutect2_multi_sample.wdl [hunk 3]
```diff
@@ -57,7 +58,8 @@ workflow Mutect2 {
 				dbsnp_index=dbsnp_index,
 				cosmic=cosmic,
 				cosmic_index=cosmic_index,
-				picard_jar=picard_jar
+				picard_jar=picard_jar,
```
Comment (LeeTL1220) [inline scripts/mutect2_wdl/mutect2_multi_sample.wdl]: Why do you need a picard jar in M2?

File: scripts/mutect2_wdl/mutect2_multi_sample.wdl [hunk 4]
```diff
@@ -68,8 +70,14 @@ workflow Mutect2 {
 	}
 
 	call CreateOutputList as filteredOutputList {
-    		input:
-    		    output_name = "filtered",
-    			vcfs = Mutect2.filtered_vcf
-    	}
+      input:
```
Comment (LeeTL1220) [inline scripts/mutect2_wdl/mutect2_multi_sample.wdl]: Indentation looks funny here.

File: scripts/mutect2_wdl/mutect2.json [hunk 5]
```diff
@@ -17,5 +17,6 @@
   "Mutect2.dbsnp_index": "$__dbsnp_index__",
```
Comment (LeeTL1220) [inline scripts/mutect2_wdl/mutect2.json]: I really like that you made two json templates.  One for single pair and one for multi-pair.

File: scripts/mutect2_wdl/mutect2_multi_sample.wdl [hunk 6]
```diff
@@ -35,6 +35,7 @@ workflow Mutect2 {
 	File cosmic
```
Comment (LeeTL1220) [inline scripts/mutect2_wdl/mutect2_multi_sample.wdl]: Can you put the description of the multi-pair tsv file format at the top of the WDL?

File: scripts/mutect2_wdl/mutect2.json [hunk 7]
```diff
@@ -17,5 +17,6 @@
...
   "Mutect2.cosmic": "$__cosmic__",
```
Comment (LeeTL1220) [inline scripts/mutect2_wdl/mutect2.json]: What happens if there is no PoN?  Are those required?  Will this work?

You should put some general usage information in the top of the M2 WDL and/or in the forum.

Comment (LeeTL1220) [inline scripts/mutect2_wdl/mutect2.json]: Imagine you are an ignorant user.

**Comment by LeeTL1220** [comment]
@davidbenjamin Also, what about an oncotator task?  Do you want to file an issue and assign to me or make it part of this pull request?  Up to you.

**Comment by davidbenjamin** [comment]
@LeeTL1220 This is for tumor-normal.  Can we discuss on Monday how to handle both tumor-only and tumor-normal with as little code duplication as possible?

**Comment by davidbenjamin** [comment]
@LeeTL1220 Filed an issue for oncotator and assigned to you.

File: scripts/mutect2_wdl/mutect2.json (hunk 1 above)
Comment (davidbenjamin) [inline scripts/mutect2_wdl/mutect2.json]: done

File: scripts/mutect2_wdl/mutect2_multi_sample.wdl (hunk 3 above)
Comment (davidbenjamin) [inline scripts/mutect2_wdl/mutect2_multi_sample.wdl]: We use `MergeVCFs` and `IntervalListTools` to scatter over intervals -- remember, this is calling the single-sample M2 (sub)-*workflow*, not the M2 *task*.

File: scripts/mutect2_wdl/mutect2_multi_sample.wdl (hunk 4 above)
Comment (davidbenjamin) [inline scripts/mutect2_wdl/mutect2_multi_sample.wdl]: fixed

File: scripts/mutect2_wdl/mutect2.json (hunk 7 above)
Comment (davidbenjamin) [inline scripts/mutect2_wdl/mutect2.json]: his will work without a PoN -- note that in wdl if `pon` is optional, then ${"--normalPanel " + pon}` yields *nothing* is `pon` is not specified.

Comment (davidbenjamin) [inline scripts/mutect2_wdl/mutect2.json]: I also made dbSNP and cosmic optional in the same way.

Comment (davidbenjamin) [inline scripts/mutect2_wdl/mutect2.json]: Added documentation to top of the wdl.

File: scripts/mutect2_wdl/mutect2.json (hunk 5 above)
Comment (davidbenjamin) [inline scripts/mutect2_wdl/mutect2.json]: Thank Takuto.

File: scripts/mutect2_wdl/mutect2_multi_sample.wdl (hunk 6 above)
Comment (davidbenjamin) [inline scripts/mutect2_wdl/mutect2_multi_sample.wdl]: done

File: scripts/mutect2_wdl/mutect2.wdl (hunk 2 above)
Comment (davidbenjamin) [inline scripts/mutect2_wdl/mutect2.wdl]: done, but do you know how to specify a workflow output that is only produced if a task in a conditional block is run?

**Comment by davidbenjamin** [comment]
@LeeTL1220 Here are the changes.  I need to run the wdl to see if they actually work.

**Comment by LeeTL1220** [comment]
Sure.  Put a time on my calendar.

**Comment by davidbenjamin** [comment]
@LeeTL1220 back to you.

File: scripts/mutect2_wdl/mutect2_multi_sample.wdl (hunk 3 above)
Comment (LeeTL1220) [inline scripts/mutect2_wdl/mutect2_multi_sample.wdl]: @davidbenjamin It's not a problem, but now you need GATK4 *and* Picard to make this workflow work.  That was definitely not assumed.  (For example, the gatk-protected docker image has no picard jar).  If you want to use the picard jar, you need to make sure the docker image has it.  

AND the GATK will soon include the picard tools, I believe.  Talk to @droazen or @lbergelson to make sure you are not out-of-date here.

TL;DR
1)  Talk to David or Louis and find out if you can call picard from GATK jar.
2)  If yes, adjust this WDL to do so.
3)  If no, make sure the picard jar is in the docker image.  Make sure to peg it to a specific version.

File: scripts/mutect2_wdl/mutect2.wdl [hunk 8]
```diff
@@ -1,3 +1,19 @@
+#  Run Mutect 2 on a single tumor-normal pair
+#
+#  Description of inputs
+#  gatk4_jar: java jar file containing gatk 4 (protected)
```
Comment (LeeTL1220) [inline scripts/mutect2_wdl/mutect2.wdl]: You may need GATK jar and picard jar (if applicable) to be String inputs.  If you put them as files, then you cannot run them through a docker image on local backends in cromwell.

Comment (LeeTL1220) [inline scripts/mutect2_wdl/mutect2.wdl]: We can discuss, if you'd like.

**Comment by LeeTL1220** [comment]
@davidbenjamin Only a couple more minor issues.

**Comment by davidbenjamin** [comment]
@LeeTL1220 It only needs a gatk jar now.  Back to you.

File: scripts/mutect2_wdl/mutect2.wdl (hunk 8 above)
Comment (davidbenjamin) [inline scripts/mutect2_wdl/mutect2.wdl]: done
